From d75d58c8a22b1132b2b2a7665c95e6ee9574baca Mon Sep 17 00:00:00 2001
From: Gil Montag <gil.montag@intel.com>
Date: Thu, 2 Jul 2020 18:47:59 +0300
Subject: [PATCH] 
 Added-INTEL_FAM6_ICELAKE_NNPI-to-intel-family.h-and-intel_rapl.c

---
 arch/x86/include/asm/intel-family.h  | 1 +
 drivers/powercap/intel_rapl_common.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/x86/include/asm/intel-family.h b/arch/x86/include/asm/intel-family.h
index 8f1e94f29a16..0a54c01b137c 100644
--- a/arch/x86/include/asm/intel-family.h
+++ b/arch/x86/include/asm/intel-family.h
@@ -75,6 +75,7 @@
 #define INTEL_FAM6_KABYLAKE_L		0x8E
 #define INTEL_FAM6_KABYLAKE		0x9E
 
+#define INTEL_FAM6_ICELAKE_NNPI         0x9D /* Icelake Neural NW processor */
 #define INTEL_FAM6_CANNONLAKE_L		0x66
 
 #define INTEL_FAM6_ICELAKE_X		0x6A
diff --git a/drivers/powercap/intel_rapl_common.c b/drivers/powercap/intel_rapl_common.c
index eb328655bc01..e040b4ff0a1c 100644
--- a/drivers/powercap/intel_rapl_common.c
+++ b/drivers/powercap/intel_rapl_common.c
@@ -992,6 +992,7 @@ static const struct x86_cpu_id rapl_ids[] __initconst = {
 	X86_MATCH_INTEL_FAM6_MODEL(ATOM_TREMONT_D,	&rapl_defaults_core),
 	X86_MATCH_INTEL_FAM6_MODEL(ATOM_TREMONT_L,	&rapl_defaults_core),
 
+	X86_MATCH_INTEL_FAM6_MODEL(ICELAKE_NNPI,        &rapl_defaults_core),
 	X86_MATCH_INTEL_FAM6_MODEL(XEON_PHI_KNL,	&rapl_defaults_hsw_server),
 	X86_MATCH_INTEL_FAM6_MODEL(XEON_PHI_KNM,	&rapl_defaults_hsw_server),
 	{}
-- 
2.17.1

